File: library/actor-common/src/main/java/im/actor/model/entity/Sex.java
/*
 * Copyright (C) 2015 Actor LLC. <https://actor.im>
 */

package im.actor.model.entity;

public enum Sex {
    UNKNOWN(1), MALE(2), FEMALE(3);

    private int value;

    Sex(int value) {
        this.value = value;
    }

    public int getValue() {
        return value;
    }

    public static Sex fromValue(int value) {
        switch (value) {
            default:
            case 1:
                return UNKNOWN;
            case 2:
                return MALE;
            case 3:
                return FEMALE;
        }
    }
}

File: actor-server/actor-models/src/main/scala/im/actor/server/model/Sex.scala
package im.actor.server.model

sealed trait Sex {
  def toOption: Option[Sex]
  def toInt: Int
}

@SerialVersionUID(1L)
case object NoSex extends Sex {
  val toOption = None
  val toInt = 1
}

@SerialVersionUID(1L)
case object Male extends Sex {
  val toOption = Some(Male)
  val toInt = 2
}

@SerialVersionUID(1L)
case object Female extends Sex {
  val toOption = Some(Female)
  val toInt = 3
}

object Sex {
  def fromInt(i: Int): Sex = i match {
    case 1 ⇒ NoSex
    case 2 ⇒ Male
    case 3 ⇒ Female
  }
}
